lib/src/wso2/wso2-api/handler/wso2-v1.ts

```diff
@@ -138,47 +137,90 @@ export const createUpdateAndPublishApiInWso2 = async (
     args.retryOptions.mutationRetries,
   );
 
-  console.log('');
-  console.log(`>>> Change api status to 'PUBLISHED'...`);
-  // will retry changing to PUBLISHED if fails
-  const endpointUrl = await backOff(async () =>
-    publishApiInWso2AndCheck({
-      wso2Axios: args.wso2Axios,
-      wso2ApiId,
-      apiDefinition: args.apiDefinition,
-      wso2Tenant: args.wso2Tenant,
-      retryOptions: args.retryOptions,
-    }),
-  );
+  if (args.lifecycleStatus) {
+    console.log('');
+    console.log(`>>> Changing lifecycle status to '${args.lifecycleStatus}'...`);
+    // will retry changing to PUBLISHED if fails
+    await backOff(async () =>
+      changeLifecycleStatusInWso2AndCheck({
+        wso2Axios: args.wso2Axios,
+        wso2ApiId,
+        apiDefinition: args.apiDefinition,
+        wso2Tenant: args.wso2Tenant,
+        retryOptions: args.retryOptions,
+        lifecycleStatus: args.lifecycleStatus,
+      }),
+    );
+  }
 
-  console.log('API created/updated and published successfully on WSO2 server');
+  // get endpoint url
+  console.log(`Getting API endpoint url`);
+  const apir = await args.wso2Axios.get(`/api/am/store/v1/apis/${wso2ApiId}`);
+
+  // find the endpoint URL of the environment that was defined in this API
+  const apid = apir.data as DevPortalAPIv1;
+  const endpointUrl = apid.endpointURLs?.reduce((acc, elem) => {
+    if (
+      elem.environmentName &&
+      args.apiDefinition.gatewayEnvironments?.includes(elem.environmentName)
+    ) {
+      if (elem.URLs?.https) {
+        return elem.URLs?.https;
+      }
+      if (elem.defaultVersionURLs?.https) {
+        return elem.defaultVersionURLs?.https;
+      }
+    }
+    return acc;
+  }, '');
+
+  console.log('API created/updated successfully on WSO2 server');
 
   return { wso2ApiId, endpointUrl };
 };
 
-export const publishApiInWso2AndCheck = async (args: {
-  wso2Axios: AxiosInstance;
-  apiDefinition: Wso2ApiDefinitionV1;
-  wso2ApiId: string;
-  wso2Tenant: string;
-  retryOptions: RetryOptions;
-}): Promise<string | undefined> => {
-  console.log(`Changing API status to PUBLISHED in WSO2`);
+const actionMap = {
+  PUBLISHED: 'Publish',
+  CREATED: 'Demote to created',
+  DEPRECATED: 'Deprecate',
+  BLOCKED: 'Block',
+  RETIRED: 'Retire',
+  PROTOTYPED: 'Deploy as a Prototype',
+};
+
+export const changeLifecycleStatusInWso2AndCheck = async (
+  args: Pick<Wso2ApiProps, 'lifecycleStatus'> &
+    Required<Pick<Wso2ApiProps, 'retryOptions' | 'apiDefinition'>> & {
+      wso2Axios: AxiosInstance;
+      wso2ApiId: string;
+      wso2Tenant: string;
+    },
+): Promise<undefined> => {
+  if (!args.lifecycleStatus) throw new Error(`'lifecycleStatus' is required`);
```

Comment:
shouldn't lifecycle status be set to "publish" by default?

Reply:
I made it optional so if it's not defined we don't touch on the lifecycle actions and leave it as it is. Another way would be to make it by default be "PUBLISHED" indeed so we would always call the lifecycle to something. What do you think?

Reply:
@MarcioMeier

Reply:
I think I prefer to not having default value here, so we just trigger the lifecycle when the caller defines it.

how would we make to not trigger the lifecycle if the default value is "publish"?

_p.s. as of my understanding, we want give the option to the user to not touch the api lifecycle in case of manual workflows_

Reply:
That was exactly the original idea 😊

Reply:
But by default, there are no manual workflows and the lifecycle should be "PUBLISH" right? Only in specific scenarios that a manual workflow is needed (in e.g. upper environments like production). 

To make the consumers' life easier setting it by default would be nicer.
But I agree, it does not make it very explicit and it might be disregarded. So anyway works for me. 😬